### DISCUSSION TITLE
Aptos Cli - adding custom static analysis to move prover.

### DISCUSSION THREAD

**Discussion by lucagiacometti19** [aptos-labs/aptos-developer-discussions#209] [Questions] [answered]
### Describe your question in detail.

I'm trying to add a custom static analysis for the move prover. 
I've successfully done it before with Diem Move (the archived repo), but I can't find a way to do so with aptos cli. I added the flag to run this custom analysis [here](https://github.com/aptos-labs/aptos-core/blob/main/third_party/move/move-prover/src/cli.rs#L145), but building the repo and running `./aptos move prove -- -h` returns `error: unexpected argument '-h' found`, which is weird since that's how I'm running commands from the old Diem Move cli (the code seems to be the same in aptos). 
To be fair I can't find a way to run most of the commands that i see declared in this clap command definition...
This is what I've added:
```
let cli = Command::new("mvp")
            .version("0.1.0")
            .about("The Move Prover")
            .author("The Diem Core Contributors")
            ...
            ...
            ...
            .arg(
                Arg::new("confidentiality")
                    .long("confidentiality")
                    .action(SetTrue)
                    .help("runs the confidentiality analysis instead of the prover.")
            )
```
Maybe I got it all wrong and the aptos cli is defined somewhere else and overrides the prover cli?

### What have you tried or looked at?  Or how can we reproduce the error?

Tried to change the prover clap command definition in [here](https://github.com/aptos-labs/aptos-core/blob/main/third_party/move/move-prover/src/cli.rs#L145).

### Which operating system are you using?

Linux (Ubuntu, Fedora, Windows WSL, etc.)

### Which SDK or tool are you using? (if any)

Aptos CLI

**Comment by gregnazario** [comment, marked as answer]
Yeah, the Aptos CLI wraps the prover with a different interface.  The original single character args were very difficult to keep track of with the 50+ commands that were there.

Here's where the Aptos CLI prover arguments are https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/src/prover.rs#L20-L89

**Comment by lucagiacometti19** [comment]
Thank you!
